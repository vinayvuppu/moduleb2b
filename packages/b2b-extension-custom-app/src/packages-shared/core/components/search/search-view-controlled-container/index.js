export { default } from './search-view-controlled-container';
