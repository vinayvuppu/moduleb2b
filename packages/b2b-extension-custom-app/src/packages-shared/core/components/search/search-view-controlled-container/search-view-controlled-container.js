import PropTypes from 'prop-types';
import React from 'react';
import SearchView from '../search-view';

export class SearchViewControlledContainer extends React.Component {
  static displayName = 'SearchViewControlledContainer';

  static propTypes = {
    value: PropTypes.object.isRequired,
    onChange: PropTypes.func.isRequired,
  };

  handleChange = nextSearchOptions => {
    this.props.onChange(nextSearchOptions);
  };

  /**
   * NOTE:
   *   When using the router container fetching is handled by the
   *   parent's lifecycle. E.g. a Discount Code list re-renders (receives new props)
   *   and performes fetching via e.g. Apollo. Loading state is then handled up
   *   the tree and not within the `SearchView`. The `onSearch` prop is meant to
   *   return a promise as defined in the `SearchView`. If a parent needs to
   *   be informed about changes in `searchOptions` use `onChange`.
   */
  handleOnSearch = () => {};

  render() {
    return (
      <SearchView
        {...this.props}
        onSearch={this.handleOnSearch}
        setSearchFiltersState={this.handleChange}
        {...this.props.value}
      />
    );
  }
}

export default SearchViewControlledContainer;
