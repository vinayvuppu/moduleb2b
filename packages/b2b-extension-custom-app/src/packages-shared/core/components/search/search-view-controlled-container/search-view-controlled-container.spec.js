import React from 'react';
import { shallow } from 'enzyme';
import SearchView from '../search-view';
import { SearchViewControlledContainer } from './search-view-controlled-container';

const createTestProps = customProps => ({
  value: {
    foo: 'foo',
    bar: 'bar',
  },
  onChange: jest.fn(),
  children: jest.fn(),

  ...customProps,
});

describe('rendering', () => {
  let props;
  let wrapper;

  beforeEach(() => {
    props = createTestProps();
    wrapper = shallow(<SearchViewControlledContainer {...props} />);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a `<SearchView>`', () => {
    expect(wrapper).toRender(SearchView);
  });

  describe('`<SearchView>`', () => {
    let searchViewWrapper;

    beforeEach(() => {
      searchViewWrapper = wrapper.find(SearchView);
    });

    it('should receive `onSearch`', () => {
      expect(searchViewWrapper).toHaveProp(
        'onSearch',
        wrapper.instance().handleOnSearch
      );
    });

    it('should receive `setSearchFiltersState`', () => {
      expect(searchViewWrapper).toHaveProp(
        'setSearchFiltersState',
        wrapper.instance().handleChange
      );
    });

    it('should receive all `value` props', () => {
      expect(searchViewWrapper).toHaveProp(
        'value',
        expect.objectContaining(props.value)
      );
    });
  });
});
