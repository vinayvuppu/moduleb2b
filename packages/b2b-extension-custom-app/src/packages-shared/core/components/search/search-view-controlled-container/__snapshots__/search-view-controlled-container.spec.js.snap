// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering should match snapshot 1`] = `
<injectTracking(withPendingRequests(SearchView))
  bar="bar"
  foo="foo"
  onChange={[MockFunction]}
  onSearch={[Function]}
  setSearchFiltersState={[Function]}
  value={
    Object {
      "bar": "bar",
      "foo": "foo",
    }
  }
>
  <Component />
</injectTracking(withPendingRequests(SearchView))>
`;
